src/server/worker/driver/driver.go

```diff
@@ -429,23 +421,16 @@ func (d *driver) RunUserErrorHandlingCode(
 	if err != nil {
 		return errors.EnsureStack(err)
 	}
-	// A context w a deadline will successfully cancel/kill
-	// the running process (minus zombies)
-	state, err := cmd.Process.Wait()
-	if err != nil {
-		return errors.EnsureStack(err)
-	}
-	if common.IsDone(ctx) {
-		if err = ctx.Err(); err != nil {
+	select {
```

Comment:
Same story.  I don't know why we do this.  The "minus zombies" comment is interesting, we need to do more testing around that (and properly act as PID 1).